recipes/foxglove-schemas-protobuf/all/patches/0001-Add-cmakelists.txt-for-building-with-Conan.patch

```diff
@@ -0,0 +1,45 @@
+From ce4386a75e077adfd04be76aaa3e98f54ed8e657 Mon Sep 17 00:00:00 2001
+From: Kevin Lannen <kevin@luci.com>
+Date: Thu, 25 May 2023 15:30:05 -0600
+Subject: [PATCH 1/2] Add cmakelists.txt for building with Conan
+
+---
+ CMakeLists.txt | 25 +++++++++++++++++++++++++
+ 1 file changed, 25 insertions(+)
+ create mode 100644 CMakeLists.txt
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+new file mode 100644
+index 0000000..c5ba3e8
+--- /dev/null
++++ b/CMakeLists.txt
+@@ -0,0 +1,25 @@
++cmake_minimum_required(VERSION 3.15)
++project(Foxglove-Schemas CXX)
++
++find_package(Protobuf 3 REQUIRED)
++
++FILE(GLOB all_protos "schemas/proto/foxglove/*.proto")
++
++FOREACH(f ${all_protos})
++    file(RELATIVE_PATH f ${CMAKE_CURRENT_SOURCE_DIR}/schemas/proto ${f})
++    STRING(REGEX REPLACE "\\.proto$" "" f ${f})
++    LIST(APPEND proto_sources "autogenerated/${f}.pb.h")
++    LIST(APPEND proto_sources "autogenerated/${f}.pb.cc")
++ENDFOREACH(f)
++
++add_custom_command(
++    OUTPUT ${proto_sources}
++    COMMAND ${CMAKE_COMMAND} -E make_directory autogenerated
++    COMMAND ${Protobuf_PROTOC_EXECUTABLE} --proto_path=${CMAKE_CURRENT_SOURCE_DIR}/schemas/proto --cpp_out=autogenerated ${all_protos}
++)
++
++add_library(foxglove_schemas_protobuf ${proto_sources})
++target_include_directories(foxglove_schemas_protobuf PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/autogenerated)
++target_link_libraries(foxglove_schemas_protobuf PUBLIC protobuf::libprotobuf)
++set_property(TARGET foxglove_schemas_protobuf PROPERTY CXX_STANDARD 17)
++install(TARGETS foxglove_schemas_protobuf)
```

Comment:
For easier maintainability I would suggest having `cmake.install()` also handle the logic of copying the relevant .pb.h files, or just disregard it altogether and copy also the library in the `package()` method - doing both seems a bit split when the things are meant to "travel" together

Reply:
This is fixed so headers and the proto files are installed in the cmake now.